### PR TITLE
fix(TextBlock): [WASM] TextBlock.Inlines are not formatted correctly: spaces/blanks are removed #11418

### LINKED ISSUE #11418: [WASM] TextBlock.Inlines are not formatted correctly: spaces/blanks are removed
### Current behavior

Please see attached repo: spaces/blanks are formatted away.

[Inlines.zip](https://github.com/unoplatform/uno/files/10802198/Inlines.zip)

### Expected behavior

WASM should work as WindowsAppSDK

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by TopProgrammer77**
GitHub Issue (If applicable): closes #11418

## PR Type

What kind of change does this PR introduce?

- Bugfix
## What is the current behavior?

https://github.com/unoplatform/uno/issues/11418

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by Youssef1313 — COMMENTED** [review]
This needs a test.

**Comment by TopProgrammer77** [comment]
I edited the commit message. But how do I add the test?

> This needs a test. Would you be able to add one please?
> 
> As for the conventional commits failure, it's because there is no space after `fix(TextBlock):`
> 
> fwiw, the issue isn't TextBlock-specific or Wasm-specific, so while fixing the conventional commit failure, consider adjusting the commit message.

**Comment by Youssef1313** [comment]
@TopProgrammer77 Probably similar to https://github.com/unoplatform/uno/blob/47e9ccb4dbbd9c2b8d833fa59998d6030a900a14/src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs#L53

**Comment by TopProgrammer77** [comment]
I added the Test

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs [hunk 1]
```diff
@@ -69,7 +70,33 @@ public void When_LineBreak_SurroundingWhiteSpace()
...
+
+
+			var tbPreserve = page.tbPreserve;
+			Assert.AreEqual(" ", tbPreserve.Inlines[1].GetText());
+			Assert.AreEqual("  ", tbPreserve.Inlines[3].GetText());
+			Assert.AreEqual(" LeftSpace", tbPreserve.Inlines[5].GetText());
+			Assert.AreEqual("RightSpace ", tbPreserve.Inlines[7].GetText());
+			Assert.AreEqual(" SurroundSpace ", tbPreserve.Inlines[9].GetText());
+			Assert.AreEqual("Middle Space", tbPreserve.Inlines[11].GetText());
+			Assert.AreEqual("  DoubleSpace  ", tbPreserve.Inlines[13].GetText());
```
Comment (Youssef1313) [inline src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs:96]: Can you please assert all inlines for clarity?

File: src/SourceGenerators/System.Xaml/System.Xaml/XamlXmlReader.cs [hunk 2]
```diff
@@ -843,12 +843,17 @@ IEnumerable<XamlXmlNodeInfo> ReadCollectionItems (XamlType parentType, XamlMembe
 				}
 
 				var currentNodeType = r.NodeType;
+				var currentNodeValue = r.Value;
 				var nextNodeType = r.MoveToContent();
-
+
```
Comment (Youssef1313) [inline src/SourceGenerators/System.Xaml/System.Xaml/XamlXmlReader.cs:848]:
```suggestion
```

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs [hunk 3]
```diff
@@ -53,6 +53,7 @@ public void When_Changing_Foreground_Property()
 		public void When_LineBreak_SurroundingWhiteSpace()
 		{
 			var page = new LineBreak_Surrounding_WhiteSpace();
+
```
Comment (Youssef1313) [inline src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs:56]:
```suggestion
```

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs [hunk 4]
```diff
@@ -69,7 +70,33 @@ public void When_LineBreak_SurroundingWhiteSpace()
...
+			Assert.AreEqual("  ", tbPreserve.Inlines[3].GetText());
+			Assert.AreEqual(" LeftSpace", tbPreserve.Inlines[5].GetText());
+			Assert.AreEqual("RightSpace ", tbPreserve.Inlines[7].GetText());
+			Assert.AreEqual(" SurroundSpace ", tbPreserve.Inlines[9].GetText());
+			Assert.AreEqual("Middle Space", tbPreserve.Inlines[11].GetText());
+			Assert.AreEqual("  DoubleSpace  ", tbPreserve.Inlines[13].GetText());
+		}
 	}
 
+
```
Comment (Youssef1313) [inline src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs:100]:
```suggestion
```

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs (hunk 1 above)
Comment (Youssef1313) [inline src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs:96]:
```suggestion
			var page = new WhiteSpace_In_TextBlock_Inlines();
			var tbDefault = page.tbDefault;
			Assert.AreEqual("Word1", tbDefault.Inlines[0].GetText());
			Assert.AreEqual(" ", tbDefault.Inlines[1].GetText());
			Assert.AreEqual("Word2", tbDefault.Inlines[2].GetText());
			Assert.AreEqual(" ", tbDefault.Inlines[3].GetText());
			Assert.AreEqual("Word3", tbDefault.Inlines[4].GetText());
			Assert.AreEqual(" LeftSpace", tbDefault.Inlines[5].GetText());
			Assert.AreEqual("Word4", tbDefault.Inlines[6].GetText());
			Assert.AreEqual("RightSpace ", tbDefault.Inlines[7].GetText());
			Assert.AreEqual("Word5", tbDefault.Inlines[8].GetText());
			Assert.AreEqual(" SurroundSpace ", tbDefault.Inlines[9].GetText());
			Assert.AreEqual("Word6", tbDefault.Inlines[10].GetText());
			Assert.AreEqual("Middle Space", tbDefault.Inlines[11].GetText());
			Assert.AreEqual("Word7", tbDefault.Inlines[12].GetText());
			Assert.AreEqual(" DoubleSpace", tbDefault.Inlines[13].GetText());

			var tbPreserve = page.tbPreserve;
			Assert.AreEqual("Word1", tbPreserve.Inlines[0].GetText());
			Assert.AreEqual(" ", tbPreserve.Inlines[1].GetText());
			Assert.AreEqual("Word2", tbPreserve.Inlines[2].GetText());
			Assert.AreEqual("  ", tbPreserve.Inlines[3].GetText());
			Assert.AreEqual("Word3", tbPreserve.Inlines[4].GetText());
			Assert.AreEqual(" LeftSpace", tbPreserve.Inlines[5].GetText());
			Assert.AreEqual("Word4", tbPreserve.Inlines[6].GetText());
			Assert.AreEqual("RightSpace ", tbPreserve.Inlines[7].GetText());
			Assert.AreEqual("Word5", tbPreserve.Inlines[8].GetText());
			Assert.AreEqual(" SurroundSpace ", tbPreserve.Inlines[9].GetText());
			Assert.AreEqual("Word6", tbPreserve.Inlines[10].GetText());
			Assert.AreEqual("Middle Space", tbPreserve.Inlines[11].GetText());
			Assert.AreEqual("Word7 ", tbPreserve.Inlines[12].GetText());
			Assert.AreEqual("  DoubleSpace  ", tbPreserve.Inlines[13].GetText());
```

**Review by Youssef1313 — COMMENTED** [review]
Minor formatting fixes and asserting all inlines suggestions. (If you will accept the suggestions directly via GitHub, don't forget to use conventional commit message)

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs (hunk 1 above)
Comment (TopProgrammer77) [reply src/Uno.UI.Tests/Windows_UI_XAML_Controls/TextBlockTests/Given_TextBlock.cs]: I finished the commit of change

**Review by Youssef1313 — APPROVED** [review]
Last time I debugged, my conclusion was that these spaces are ignored by `System.Xml.XmlReader` itself, but as long as the test pass, I think my conclusion was wrong. So LGTM.

@jeromelaban for a second review.

**Comment by jeromelaban** [comment]
This is looking good to me as well! Thanks for your contribution!

Could you sign the CLA as well (see https://github.com/unoplatform/uno/pull/11929#issuecomment-1500004315) ?

**Comment by TopProgrammer77** [comment]
I've signed the CLA.

**Pull request merged**